scripts/gen_dockerfile_test.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python3
+
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit test for gen_dockerfile.py"""
+
+import argparse
+import filecmp
+import os
+import shutil
+import subprocess
+import unittest.mock
+
+import pytest
+import yaml
+
+import gen_dockerfile
+
+
+# Expected list of files generated
+EXPECTED_OUTPUT_FILES = ['Dockerfile', '.dockerignore']
+
+
+@pytest.fixture
+def testdata_dir():
+    testdata_dir = os.path.join(os.path.dirname(__file__), 'testdata')
+    assert os.path.isdir(testdata_dir), (
+        'Could not run test: testdata directory not found')
+    return testdata_dir
+
+
+def compare_file(filename, dir1, dir2):
+    """Compare identically named files in two different directories"""
+    assert filecmp.cmp(
+        os.path.join(dir1, filename), os.path.join(dir2, filename))
+
+
+@pytest.mark.parametrize('app_yaml, isfile, expected', [
+    # Basic app.yaml
+    ('env: flex', False, {
+        'base_image': 'some_image_name',
+        'dockerfile_python_version': '',
+        'has_requirements_txt': False,
+        'entrypoint': '',
+    }),
+    # All supported python versions
+    ('runtime_config:\n python_version:', False, {
+        'dockerfile_python_version': '',
+    }),
+    ('runtime_config:\n python_version: 2', False, {
+        'dockerfile_python_version': '',
+    }),
+    ('runtime_config:\n python_version: 3', False, {
+        'dockerfile_python_version': '3.5',
+    }),
+    ('runtime_config:\n python_version: 3.4', False, {
+        'dockerfile_python_version': '3.4',
+    }),
+    ('runtime_config:\n python_version: 3.5', False, {
+        'dockerfile_python_version': '3.5',
+    }),
+    # requirements.txt present
+    ('env: flex', True, {
+        'has_requirements_txt': True,
```

Comment:
No action needed, but generally if your parametrization has an argument that is the same value except for one case it's a good idea to pull it into a separate test to simplify the parametrization and make the test case more straightforward.